### DISCUSSION TITLE
Pedantic: BrowserKit.String.remove(_ string: String?) is poorly named

### DISCUSSION THREAD

**Discussion by gutley** [mozilla-mobile/firefox-ios#13272] [General]
Because:

1.  It's using the naming convention of a mutating func. Since it isn't one , it should probably be `removing(_ string: String?)` to make it clear its a non-mutating-returning func. (But see 3)
2. There are already several other mutating funcs in String named `remove` or `removeXXX` so this one's differing semantics is confusing.
3. It's not immediately clear from the call site what it's doing. Since its just a wrapper on `replacingOccurrences(of:)` maybe a clearer name would be `removingOccurences(of:)`

Also, not keen on it accepting an optional string. None of the normal string manipulation funcs in stdLib behave that way and none of its existing call sites can ever pass in a nil parameter anyway.

I did say it was pedantic 😀

**Comment by lmarceau** [comment]
Thanks @gutley, yeah those points totally make sense. I moved this extension from the Shared target into BrowserKit on PR #12718. If you wish to make changes to improve the naming, I'll gladly review the PR. Thank you!

**Reply by gutley** [reply]
@lmarceau PR here: https://github.com/mozilla-mobile/firefox-ios/pull/13318
